pkg/osquery/tables/gsettings/metadata.go

```diff
@@ -0,0 +1,283 @@
+// +build !windows
+
+package gsettings
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	osquery "github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+type GsettingsMetadata struct {
+	client    *osquery.ExtensionManagerClient
+	logger    log.Logger
+	cmdRunner func(ctx context.Context, args []string, tmpdir string, output *bytes.Buffer) error
+}
+
+// Metadata returns a table plugin for querying metadata about specific keys in
+// specific schemas
+func Metadata(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+
+	columns := []table.ColumnDefinition{
+		// TODO: maybe need to add 'path' for relocatable schemas..
+		table.TextColumn("schema"),
+		table.TextColumn("key"),
+		table.TextColumn("description"),
+		table.TextColumn("type"),
+	}
+
+	t := &GsettingsMetadata{
+		client:    client,
+		logger:    logger,
+		cmdRunner: execGsettingsCommand,
+	}
+
+	return table.NewPlugin("kolide_gsettings_metadata", columns, t.generate)
+}
+
+func (t *GsettingsMetadata) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	schemas := tablehelpers.GetConstraints(queryContext, "schema", tablehelpers.WithAllowedCharacters(allowedCharacters))
+	if len(schemas) < 1 {
+		return results, errors.New("kolide_gsettings_metadata table requires at least one schema to be specified")
+	}
+
+	for _, schema := range schemas {
+		descriptions, err := t.gsettingsDescribeForSchema(ctx, schema)
+		if err != nil {
+			level.Info(t.logger).Log(
+				"msg", "error describing keys for schema",
+				"schema", schema,
+				"err", err,
+			)
+			continue
+		}
+		for _, d := range descriptions {
+			row := map[string]string{
+				"description": d.Description,
+				"type":        d.Type,
+				"schema":      schema,
+				"key":         d.Key,
+			}
+			results = append(results, row)
+		}
+
+	}
+
+	return results, nil
+}
+
+type keyDescription struct {
+	Description string
+	Type        string
+	Key         string
+}
+
+func (t *GsettingsMetadata) gsettingsDescribeForSchema(ctx context.Context, schema string) ([]keyDescription, error) {
+	var descriptions []keyDescription
+
+	dir, err := ioutil.TempDir("", fmt.Sprintf("osq-gsettings-metadata-%s", schema))
```

Comment:
(nit) you don't need all that extra stuff in the temp dir name. These are short lived, and just to catch any kinda weird behavior. `osq-gsettings` seems fine.